R/turns.R

```diff
@@ -35,13 +38,27 @@ Turn <- new_class(
     ),
     role = new_property(
       class = class_character,
-      getter = function(self) "unknown"
+      getter = function(self) self@role %||% "unknown"
     )
   ),
-  constructor = function(contents = list()) {
+  constructor = function(role = "", contents = list()) {
     if (is.character(contents)) {
       contents <- list(ContentText(paste0(contents, collapse = "\n")))
     }
+
+    if (is_string(role) && nzchar(role)) {
+      if (identical(role, "user")) {
+        return(UserTurn(contents = contents))
+      }
+      if (identical(role, "assistant")) {
+        return(AssistantTurn(contents = contents))
+      }
+      if (identical(role, "system")) {
+        return(SystemTurn(contents = contents))
+      }
+      return(new_object(S7_object(), contents = contents, role = role))
```

Comment:
I think you could still error here?

Reply:
Updated to throw for unexpected roles here: 1a3d3fbe7730b1256c929417fa7116499c353d9d